lib/puppet/util/windows/adsi.rb

```diff
@@ -107,13 +107,42 @@ def execquery(query)
   end
 
   module Shared
+    def local_names
+      @local_names ||= [
+        # localized version of BUILTIN
+		# for instance VORDEFINIERT on German Windows
+        Puppet::Util::Windows::SID.sid_to_name('S-1-5-32').upcase,
+        # localized version of NT AUTHORITY (can't use S-1-5)
+		# for instance AUTORITE NT on French Windows
+        Puppet::Util::Windows::SID.name_to_sid_object('SYSTEM').domain.upcase		
+	  ]
+	end
     def uri(name, host = '.')
-      host = '.' if ['NT AUTHORITY', 'BUILTIN', Socket.gethostname].include?(host)
+      # require 'pry'; binding.pry
+      host = '.' if (local_names << Socket.gethostname).include?(host.upcase)
```

Comment:
@Iristyle per discussion, need to change to `Socket.gethostname.upcase` to avoid

```
 1) Puppet::Util::Windows::ADSI Puppet::Util::Windows::ADSI::Group an instance should generate the correct URI
    Failure/Error: expect(group.uri).to eq("WinNT://./#{groupname},group")

      expected: "WinNT://./testgroup,group"
           got: "WinNT://testcomputername/testgroup,group"

      (compared using ==)
    # ./spec/unit/util/windows/adsi_spec.rb:419:in `block (4 levels) in <top (required)>'
```